subworkflows/local/align_cellrangermulti.nf

```diff
@@ -97,7 +97,12 @@ workflow CELLRANGER_MULTI_ALIGN {
                     lines = [ 'sample_id,cmo_ids,description' ]
                     meta[0].each{ lines = lines + [ "$it.multiplexed_sample_id,$it.cmo_ids,$it.description" ] }
                     cmos = file( "${workflow.workDir}/${sample}_cmo_samplesheet.csv" )
-                    cmos.text = lines.join("\n").trim().toString()
+                    if (
+                        cmos.exists() &&
+                        cmos.text == lines.join("\n").trim().toString()
```

Comment:
this line does nothing, should this be an assert statement?

Reply:
Just wanted to add a checker so that, if the file already exists and has the same contents, indeed nothing happens and it is just loaded to allow `caching`. But, if anything else, we populate the new contents.

See [here](https://github.com/nf-core/scrnaseq/pull/276#discussion_r1559349855) as well.

Reply:
Not relevant anymore as it was migrated to a python script as a module.